### ISSUE TITLE
Generate PNG images directly

### ISSUE THREAD

**Issue by jdfrens** [jdfrens/fractals_elixir#6]
Using `imagemagick` to convert from PPM to PNG is an interesting problem, but I'm sick of it now.  I don't like the output flow this way, and I'd rather solve the problem of writing a PNG (or any other format).

Images are written in either PPM or PNG formats.

This will make it easier to write outputs for Phoenix channels (JavaScript canvases?) and Scenic windows.

* https://www.w3.org/TR/2003/REC-PNG-20031110/ (the standard)
* https://github.com/yuce/png (Erlang library to write some PNG variations)

**Comment by jdfrens** [comment]
tl;dr... reimplement the `png` library

I would like to use a library, but I can't get [`png`](https://github.com/yuce/png) to work.  For some reason, pattern matching on charlists seems to be broken:

```elixir
# Elixir call
:png.chunk('IEND')
```

```erlang
# Erlang definition
chunk('IEND') ->
    chunk(<<"IEND">>, <<>>).
```

Result: 

```
** (FunctionClauseError) no function clause matching in :png.chunk/1

    The following arguments were given to :png.chunk/1:

        # 1
        'IEND'

    (png) src/png.erl:161: :png.chunk/1
    silly.exs:21: Foo.doit/0
    (elixir) lib/code.ex:767: Code.require_file/2
    (mix) lib/mix/tasks/run.ex:146: Mix.Tasks.Run.run/5
```

I've tried everything I could think of.  I'm obviously getting into the library because the stacktrace matches us just fine.  Also, I'm able to extract a usable record definition from the Erlang library into Elixir.  And given the code and result above, it's only matching on a charlist, and that seems to be failing.

The code for `png` is 3.5 years old, so I'm wondering if there's something peculiar with more recent versions of OTP that are causing an issue.  Why that should affect something as basic as pattern matching on charlists is beyond me.

I take this as a sign that the universe wants me to implement the code myself.  And I can just do a transliteration of that code.

**Comment by jdfrens** [comment]
I'm closing this even though the Issue 6 branch/commits only make it possible to write a PNG file.  The CLI is not actually hooked up to actually do this.    Issue #13 will do this, after #9 is finished.
